test/framework/src/main/java/org/elasticsearch/transport/nio/ESSelector.java

```diff
@@ -156,16 +155,8 @@ public void removeRegisteredChannel(NioChannel channel) {
 
     @Override
     public void close() throws IOException {
-        close(false);
-    }
-
-    public void close(boolean shouldInterrupt) throws IOException {
         if (isClosed.compareAndSet(false, true)) {
-            if (shouldInterrupt && thread != null) {
-                thread.interrupt();
-            } else {
-                wakeup();
-            }
+            wakeup();
             runLock.lock(); // wait for the shutdown to complete
```

Comment:
I don't like how we lock that `runLock` here. we also use it to signal that we are running. I think what we should do it to have a `CountDownLatch exitedLoop = new CountDownLatch(1);` that we can wait on conditionally. ie:

```Java

if (isRunning()) {
  exitedLoop.await();
}
```

and count it down once we exited the loop, makes sense?